react-ui/src/App.jsx

```diff
@@ -64,4 +50,47 @@ function App() {
   );
 }
 
+function ModalSwitch() {
+  const location = useLocation();
+
+  const background = location.state && location.state.background;
+
+  return (
+    <div>
+      <Routes location={background || location}>
+        <Route exact path="/" element={<Navigate replace to="home" />} />
+        <Route exact path="/authorizeCallback" element={<AuthorizeCallback />} />
+        <Route exact path="/home" element={<Home />} />
+        <Route exact path="/contests" element={<Contests />} />
+        <Route exact path="/contests/:contestId" element={<Contest />} />
+        <Route exact path="/mod/review" element={<ReviewSubmissions />} />
+        <Route exact path="/submission" element={<Submission />} />
+        <Route
+          exact
+          path="/submit"
+          element={<Navigate replace state={{ defaultTab: 1 }} to="/submission" />}
+        />
+        <Route exact path="/hallOfFame" element={<HallOfFame />} />
+        <Route exact path="/profile/settings" element={<Settings />} />
+        <Route path="*" element={<NotFound />} />
+      </Routes>
+      {background && (
```

Comment:
I just tested out the code and the entry page links are not deep-linked. If I attempt to visit `/contests/aug23/entry/gKTeXOqP` directly, I reach the 404 page. I'm guessing because the `background` variable never gets set, so the path isn't accessible.

Why is the `background` variable needed here?

Reply:
Found this example, which looks like you may have been following, https://v5.reactrouter.com/web/example/modal-gallery. I think you need to add `<Route exact path="/contests/:contestId/entry/:entryId" element={<Entry />} />` in the paths along with the modal.

Reply:
Ya good catch, I had saved it originally but didn't leave a comment to remind me why. Fix now.

Reply:
Ok, for some reason it works when logged out but not when logged in. Need to chase this up.

Reply:
Ok, `[...winners, ...entries].find((entry) => entry.id === entryId)` was returning `undefined`, which threw an error when trying to destructure. Set a default value of `{}` to correct.